flow/project.py

```diff
@@ -1660,12 +1660,19 @@ def _execute_operation(self, operation, timeout=None):
             or operation.name not in self._operation_functions
             # The specified executable is not the same as the interpreter instance:
             or operation.directives.get('executable', sys.executable) != sys.executable
+            # the operation requires mpi or omp_num_thread
+            or operation.directives.get('nranks', 1) > 1
+            or operation.directives.get('omp_num_threads', 1) > 1
         ):
             # ... need to fork:
+            if operation.directives['nranks']:
+                prefix = self._environment.get_prefix(operation)
+            else:
+                prefix = ''
```

Comment:
From Brandon:

Change `get_prefix` to automatically return an empty string if nranks is not set. Then you can get rid of the conditional logic here and remove lines 1668, 1670, and 1671.